src/Doctrine/ORM/ProductRepositoryTrait.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Loevgaard\SyliusBrandPlugin\Doctrine\ORM;
+
+use Doctrine\ORM\QueryBuilder;
+use Loevgaard\SyliusBrandPlugin\Entity\BrandInterface;
+
+trait ProductRepositoryTrait
+{
+    /**
+     * @param string $alias
+     * @param string|null $indexBy The index for the from.
+     *
+     * @return QueryBuilder
+     */
+    abstract public function createQueryBuilder($alias, $indexBy = null);
```

Comment:
In traits we should define some methods we use in trait as abstract that means that methods should be defined in classes where we use that traits. Here we say that `createQueryBuilder` should be defined because trait use that method. And when app developer extend custom repository from original `EntityRepository` and use `ProductRepositoryTrait` - all puzzles will be in one place.

Defining method in phpdoc will mean that trait contain that method (implemented), but it is not true.

Reply:
Yeah, I understand. Let's keep it this way